main/logging.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-extension-foundation/settings"
+	"github.com/pkg/errors"
+)
+
+var (
+	infoOperation    = "INFO"
+	warningOperation = "WARNING"
+	errorOperation   = "ERROR"
+)
+
+type customGeneralLogger struct {
+	logger    *log.Logger
+	operation string
+}
+
+func (l *customGeneralLogger) Print(v ...interface{}) {
+	l.logger.Print(formatLoggingMessage(l.operation, fmt.Sprint(v...)))
+}
+
+func (l *customGeneralLogger) Println(v ...interface{}) {
+	l.logger.Println(formatLoggingMessage(l.operation, fmt.Sprint(v...)))
+}
+
+func (l *customGeneralLogger) Printf(format string, v ...interface{}) {
+	l.logger.Printf(formatLoggingMessage(l.operation, fmt.Sprintf(format, v...)))
+}
+
+func formatLoggingMessage(messageType string, message string) string {
+	currentTime := time.Now().UTC().Format("2006-01-02T15:04:05.999Z")
+
+	//Caller(1) is the customlogger Print function in this file
+	//Caller(2) is the line that called the logger's print function
+	_, filePath, line, _ := runtime.Caller(2)
+
+	//Gets the line number and caller of the log message to identify what line logged
+	shortFileArr := strings.Split(filePath, "/")
+	shortFile := shortFileArr[len(shortFileArr)-1]
+
+	return fmt.Sprintf("[%s] [%s] [%s:%d] [%s]: %s", currentTime, version, shortFile, line, messageType, message)
+}
+
+type customOperationLogger struct {
+	logger *log.Logger
+}
+
+func (l *customOperationLogger) Printf(format string, v ...interface{}) {
+	l.logger.Printf(formatOperationMessage(fmt.Sprintf(format, v...)))
+}
+
+func formatOperationMessage(message string) string {
+	currentTime := time.Now().UTC().Format("2006-01-02T15:04:05.999Z")
+	return fmt.Sprintf("[%s] [%s]: %s", currentTime, version, message)
+}
+
+func initAllLogging() (generalFile *os.File, operationFile *os.File, generalErr error, operationErr error) {
```

Comment:
If something is being exported then it should start with a capital letter, else start with lower case.